src/kernels/jupyter/launcher/jupyterExecution.ts

```diff
@@ -234,12 +234,12 @@ export class JupyterExecutionBase implements IJupyterExecution {
         if (options.localJupyter) {
             // If that works, then attempt to start the server
             traceInfo(`Launching server`);
-            const useDefaultConfig = !options || options.skipUsingDefaultConfig ? false : true;
             const settings = this.configuration.getSettings(options.resource);
-
+            const useDefaultConfig = this.configuration.getSettings(options.resource).useDefaultConfigForJupyter;
+            const workingDir = await this.workspace.computeWorkingDirectory(options.resource);
             // Expand the working directory. Create a dummy launching file in the root path (so we expand correctly)
             const workingDirectory = expandWorkingDir(
-                options.workingDir,
```

Comment:
This value in the options property was totally useless, as we can see the next line we access the workspace object.
hence why not compute the workspace directory here instead of passing properties around.

This simplifies the type a lot, fewer properties.

This simplifies the type a lot